### PR TITLE
bugfix: forbidding execute SQL which update pk value

### PR DESCRIPTION
**Pull request by 81519434**
### Ⅰ. Describe what this PR did
Forbidding execute SQL which updates pk value.

### Ⅱ. Does this pull request fix one issue?
#3066

### CODE REVIEW

**Review by slievrly — APPROVED** [review]
LGTM

**Review by a364176773 — APPROVED** [review]
LGTM

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java [hunk 1]
```diff
@@ -172,4 +173,13 @@ public static boolean isLockRetryPolicyBranchRollbackOnConflict() {
             return LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT;
         }
     }
+
+    protected void isUpdatePkValue(List<String> updateColumns) {
```
Comment (jsbxyyx) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java:177]: return boolean? control of exception by caller?

Comment (81519434) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java:177]: I think this is unnecessary. It just checks if the param is legal. There is nothing to control by the caller.

Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java:177]: this method is more suitable placed in UpdateExecutor or MultiUpdateExecutor. or rename method name.

Comment (81519434) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java:177]: What method name do you want? Please give me some advice

Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java:177]: how about `assertContainsPKColumnName`?

**Review by jsbxyyx — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**